src/duckling/game/drawable/tile-block-drawable-drawer.ts

```diff
@@ -0,0 +1,114 @@
+import {RenderTexture, Texture, Sprite, Graphics, DisplayObject, Container, BaseTexture} from 'pixi.js';
+
+import {Vector} from '../../math/vector';
+import {AssetService} from '../../project/asset.service';
+
+import {TileBlockDrawable, getTileWidth} from './tile-block-drawable';
+
+export function drawTileBlockDrawable(tileBlockDrawable : TileBlockDrawable, assetService : AssetService) : DisplayObject {
+    if (!tileBlockDrawable.textureKey) {
+        return null;
+    }
+
+    let baseTexture = assetService.get({key: tileBlockDrawable.textureKey, type: "TexturePNG"});
+    if (!baseTexture) {
+        return null;
+    }
+    
+    // let container = new Container();
+    return _constructTileBlockSprite(tileBlockDrawable, baseTexture, assetService);
+    // return container;
+}
+
+function _constructTileBlockSprite(tileBlockDrawable : TileBlockDrawable, baseTexture : BaseTexture, assetService : AssetService) : DisplayObject {
+    let tileSize = getTileWidth(tileBlockDrawable, assetService);
+    if (tileSize <= 0) throw new Error("Invalid tile size specified");
+    let numberOfTiles = { x: tileBlockDrawable.size.x / tileSize, y: tileBlockDrawable.size.y / tileSize};
+    
+    let positions = _getPositions(numberOfTiles);
+    let calculateOffset: {x: boolean, y: boolean} = _getOffsets(numberOfTiles);
+    
+    return _constructTileBlockContainer(positions, calculateOffset, baseTexture, tileSize, numberOfTiles);
+}
+
+function _getPositions(numberOfTiles: Vector): Vector[] {
+    let positions: Vector[] = [];
+    
+    let position: Vector;
+    for (let xPosition of _getAxisPositions(numberOfTiles.x)) {
+        for (let yPosition of _getAxisPositions(numberOfTiles.y)) {
+            position = { x: xPosition, y: yPosition };
+            positions.push(position);
```

Comment:
The position variable is not necessary. You should push an object literal instead.